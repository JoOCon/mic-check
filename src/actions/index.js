export const addUser = (user) => ({
  type: 'ADD_USER',
  user
});

export const getLocalStorageUsers = (users) => ({
  type: 'GET_LOCAL_STORAGE_USERS',
  users
});

export const setActiveUser = (user) => ({
  type: 'SET_ACTIVE_USER',
  user
});

export const logoutActiveUser = () => ({
  type: 'LOGOUT_ACTIVE_USER'
});

export const getLocalStorageRentals = (rentals) => ({
  type: 'GET_LOCAL_STORAGE_RENTALS',
  rentals
});

export const addRental = (item) => ({
  type: 'ADD_RENTAL',
  item
});

export const removeRental = (item) => ({
  type: 'REMOVE_RENTAL',
  item
});

export const addQuote = (deliveryQuote) => ({
  type: 'ADD_QUOTE',
  deliveryQuote
});
