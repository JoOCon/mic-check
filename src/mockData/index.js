export const mockUser = {
  userName: 'Tom Hanks',
  password: 'password',
  email: 'THanks@hanks.com',
  location: '20 McAllister St, San Francisco, CA'
};

export const mockUsers = [{
  userName: 'Tom Hanks',
  password: 'password',
  email: 'THanks@hanks.com',
  location: '20 McAllister St, San Francisco, CA'
},
{
  userName: 'Jim Hanks',
  password: 'password',
  email: 'JHanks@hanks.com',
  location: '20 McAllister St, San Francisco, CA'
}];

export const mockItem = {
  name: 'Guitar',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a guitar!',
  image: "https://guitar-image.png",
  id: 1
};

export const mockItem2 = {
  name: 'Amp',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a amp!',
  image: "https://amp-image.png",
  id: 2
};

export const mockItems = [{
  name: 'Guitar',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a guitar!',
  image: "https://guitar-image.png",
  id: 1
}, 
{
  name: 'Amp',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a amp!',
  image: "https://amp-image.png",
  id: 2
}];

export const mockSenderLocation = '101 Market St, San Francisco, CA';
export const mockDeliveryLocation = '20 McAllister St, San Francisco, CA';

export const mockQuoteData = {
  kind: "delivery_quote",
  fee: 1140,
  created: "2018-09-13T00:40:02Z",
  pickup_duration: 5,
  expires: "2018-09-13T00:45:01Z",
  currency: "usd",
  currency_type: "USD",
  duration: 45,
  dropoff_eta: "2018-09-13T01:25:02Z",
  id: "dqt_Ltalsge5ktl1YV"
};

export const mockOrderInfo = {
  manifest: 'a box of shoes',
  'manifest_items[].name': 'Nikes',
  'manifest_items[].size': '2',
  pickup_address: '101 Market St, San Francisco, CA',
  pickup_name: 'Allen',
  pickup_phone_number: 3033334444,
  dropoff_address: '20 McAllister St, San Francisco, CA',
  dropoff_name: 'Tim',
  dropoff_phone_number: 2223334444,
  dropoff_notes: 'Tim@timmmy.com'
};