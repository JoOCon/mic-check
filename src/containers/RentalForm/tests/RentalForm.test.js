import React from 'react';
import { shallow } from 'enzyme';
import { RentalForm, mapStateToProps, mapDispatchToProps } from '../';
import * as mockData from '../../../mockData';
import * as actions from '../../../actions';

describe('RentalForm tests', () => {
  let wrapper;
  let mockRunFunction;
  const { mockUser, mockItem } = mockData;
  const { addRental, removeRental } = actions;
  beforeEach(() => {
    mockRunFunction = jest.fn();
    wrapper = shallow( 
      <RentalForm 
        activeUser={mockUser}
        addRental={mockRunFunction}
        removeRental={mockRunFunction}
      /> 
    );
  });

  it('renders the RentalForm with the correct props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('MapStateToProps', () => {
    it('should have a activeUser data object in props', () => {
      const mockState = {activeUser: mockUser};
      const expected = {activeUser: mockUser};
      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    });
  });

  describe('MapDispatchToProps', () => {
    let mockDispatch;
    beforeEach(() => {
      mockDispatch = jest.fn();
    });

    it('should dispatch addRental when called', () => {
      const mockAction = addRental(mockItem);
      const props = mapDispatchToProps(mockDispatch);
      props.addRental(mockItem);
      expect(mockDispatch).toBeCalledWith(mockAction);
    });

    it('should dispatch removeRental when called', () => {
      const mockAction = removeRental(mockItem);
      const props = mapDispatchToProps(mockDispatch);
      props.removeRental(mockItem);
      expect(mockDispatch).toBeCalledWith(mockAction);
    });
  });
});
