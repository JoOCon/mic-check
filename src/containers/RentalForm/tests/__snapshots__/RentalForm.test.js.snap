// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RentalForm tests renders the RentalForm with the correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RentalForm
    activeUser={
        Object {
            "email": "THanks@hanks.com",
            "location": "20 McAllister St, San Francisco, CA",
            "password": "password",
            "userName": "Tom Hanks",
          }
    }
    addRental={[Function]}
    removeRental={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          className="rental-form"
          onSubmit={[Function]}
>
          <h2
                    className="log-header"
          >
                    Manage Rental
          </h2>
          <input
                    className="input-field"
                    name="name"
                    onChange={[Function]}
                    placeholder="Item Name"
                    value=""
          />
          <input
                    className="input-field"
                    name="street"
                    onChange={[Function]}
                    placeholder="Street"
                    value=""
          />
          <input
                    className="input-field"
                    name="city"
                    onChange={[Function]}
                    placeholder="City"
                    value=""
          />
          <input
                    className="input-field"
                    name="state"
                    onChange={[Function]}
                    placeholder="State"
                    value=""
          />
          <input
                    className="input-field"
                    name="description"
                    onChange={[Function]}
                    placeholder="Item Description"
                    type="text"
                    value=""
          />
          <input
                    className="input-field"
                    name="rentalPrice"
                    onChange={[Function]}
                    placeholder="Rental Price"
                    type="number"
                    value=""
          />
          <input
                    className="input-field"
                    name="image"
                    onChange={[Function]}
                    placeholder="Image"
                    type="url"
                    value=""
          />
          <button
                    className="submit"
          >
                    Submit
          </button>
</form>,
        false,
        <Link
          replace={false}
          to="/Tom Hanks"
>
          <button
                    className="back"
          >
                    Back
          </button>
</Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="log-header"
>
              Manage Rental
</h2>,
            <input
              className="input-field"
              name="name"
              onChange={[Function]}
              placeholder="Item Name"
              value=""
/>,
            <input
              className="input-field"
              name="street"
              onChange={[Function]}
              placeholder="Street"
              value=""
/>,
            <input
              className="input-field"
              name="city"
              onChange={[Function]}
              placeholder="City"
              value=""
/>,
            <input
              className="input-field"
              name="state"
              onChange={[Function]}
              placeholder="State"
              value=""
/>,
            <input
              className="input-field"
              name="description"
              onChange={[Function]}
              placeholder="Item Description"
              type="text"
              value=""
/>,
            <input
              className="input-field"
              name="rentalPrice"
              onChange={[Function]}
              placeholder="Rental Price"
              type="number"
              value=""
/>,
            <input
              className="input-field"
              name="image"
              onChange={[Function]}
              placeholder="Image"
              type="url"
              value=""
/>,
            <button
              className="submit"
>
              Submit
</button>,
          ],
          "className": "rental-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Manage Rental",
              "className": "log-header",
            },
            "ref": null,
            "rendered": "Manage Rental",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "name",
              "onChange": [Function],
              "placeholder": "Item Name",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "street",
              "onChange": [Function],
              "placeholder": "Street",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "city",
              "onChange": [Function],
              "placeholder": "City",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "state",
              "onChange": [Function],
              "placeholder": "State",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "description",
              "onChange": [Function],
              "placeholder": "Item Description",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "rentalPrice",
              "onChange": [Function],
              "placeholder": "Rental Price",
              "type": "number",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "image",
              "onChange": [Function],
              "placeholder": "Image",
              "type": "url",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="back"
>
            Back
</button>,
          "replace": false,
          "to": "/Tom Hanks",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Back",
            "className": "back",
          },
          "ref": null,
          "rendered": "Back",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="rental-form"
            onSubmit={[Function]}
>
            <h2
                        className="log-header"
            >
                        Manage Rental
            </h2>
            <input
                        className="input-field"
                        name="name"
                        onChange={[Function]}
                        placeholder="Item Name"
                        value=""
            />
            <input
                        className="input-field"
                        name="street"
                        onChange={[Function]}
                        placeholder="Street"
                        value=""
            />
            <input
                        className="input-field"
                        name="city"
                        onChange={[Function]}
                        placeholder="City"
                        value=""
            />
            <input
                        className="input-field"
                        name="state"
                        onChange={[Function]}
                        placeholder="State"
                        value=""
            />
            <input
                        className="input-field"
                        name="description"
                        onChange={[Function]}
                        placeholder="Item Description"
                        type="text"
                        value=""
            />
            <input
                        className="input-field"
                        name="rentalPrice"
                        onChange={[Function]}
                        placeholder="Rental Price"
                        type="number"
                        value=""
            />
            <input
                        className="input-field"
                        name="image"
                        onChange={[Function]}
                        placeholder="Image"
                        type="url"
                        value=""
            />
            <button
                        className="submit"
            >
                        Submit
            </button>
</form>,
          false,
          <Link
            replace={false}
            to="/Tom Hanks"
>
            <button
                        className="back"
            >
                        Back
            </button>
</Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="log-header"
>
                Manage Rental
</h2>,
              <input
                className="input-field"
                name="name"
                onChange={[Function]}
                placeholder="Item Name"
                value=""
/>,
              <input
                className="input-field"
                name="street"
                onChange={[Function]}
                placeholder="Street"
                value=""
/>,
              <input
                className="input-field"
                name="city"
                onChange={[Function]}
                placeholder="City"
                value=""
/>,
              <input
                className="input-field"
                name="state"
                onChange={[Function]}
                placeholder="State"
                value=""
/>,
              <input
                className="input-field"
                name="description"
                onChange={[Function]}
                placeholder="Item Description"
                type="text"
                value=""
/>,
              <input
                className="input-field"
                name="rentalPrice"
                onChange={[Function]}
                placeholder="Rental Price"
                type="number"
                value=""
/>,
              <input
                className="input-field"
                name="image"
                onChange={[Function]}
                placeholder="Image"
                type="url"
                value=""
/>,
              <button
                className="submit"
>
                Submit
</button>,
            ],
            "className": "rental-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Manage Rental",
                "className": "log-header",
              },
              "ref": null,
              "rendered": "Manage Rental",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "name",
                "onChange": [Function],
                "placeholder": "Item Name",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "street",
                "onChange": [Function],
                "placeholder": "Street",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "city",
                "onChange": [Function],
                "placeholder": "City",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "state",
                "onChange": [Function],
                "placeholder": "State",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "description",
                "onChange": [Function],
                "placeholder": "Item Description",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "rentalPrice",
                "onChange": [Function],
                "placeholder": "Rental Price",
                "type": "number",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "image",
                "onChange": [Function],
                "placeholder": "Image",
                "type": "url",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="back"
>
              Back
</button>,
            "replace": false,
            "to": "/Tom Hanks",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Back",
              "className": "back",
            },
            "ref": null,
            "rendered": "Back",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
